src/project-generator.ts

```diff
@@ -89,31 +89,44 @@ export class ProjectGenerator {
     }
 
     private async updateContent(appName: string, codeSnippet: string, vspecPath: string, vspecJsonBlobSha?: string): Promise<number> {
-        const appManifestBlobSha = await this.getNewAppManifestSha(appName, vspecPath);
-        const mainPyBlobSha = await this.getNewMainPySha(appName, codeSnippet);
+        const convertedCode = await this.getConvertedCode(appName, codeSnippet);
+        const appManifestBlobSha = await this.getNewAppManifestSha(appName, vspecPath, convertedCode.dataPoints);
+        const mainPyBlobSha = await this.getNewMainPySha(convertedCode.finalizedMainPy);
 
         await this.gitRequestHandler.updateTree(appManifestBlobSha, mainPyBlobSha, vspecJsonBlobSha);
         return StatusCodes.OK;
     }
 
-    private async getNewAppManifestSha(appName: string, vspecPath: string): Promise<string> {
+    private async getConvertedCode(appName: string, codeSnippet: string): Promise<ConvertedCode> {
```

Comment:
```suggestion
    private async convertCode(appName: string, codeSnippet: string): Promise<ConvertedCode> {
```

`get` and `set` naming should be used for accessors and modifiers. Here you are invoking functionality which should always start with a verb expressing what the functionality is doing.